client/reader/user-profile/user-profile.utils.ts

```diff
@@ -1,12 +1,23 @@
 /**
  * Return the URL of the user profile page for a given username.
+ * This the preferred URL format for the user profile.
  *
  * Example: `/reader/users/user_login`
  */
-export function getUserProfileUrl( userLogin: string ): string {
+export function getUserProfileUrlWithUsername( userLogin: string ): string {
 	return `/reader/users/${ userLogin }`;
 }
 
+/**
+ * Return the URL of the user profile page for a given user ID.
+ * This an alternative URL format for the user profile.
+ *
+ * Example: `/reader/users/userId`
+ */
+export function getUserProfileUrlWithId( userId: string ): string {
+	return `/reader/users/${ userId }`;
+}
```

Comment:
These functions seem about identical, other than the input type. It seems like we could just continue using the original `getUserProfileUrl`.

Im a little concerned having the separate `withId` version might also read a bit confusing considering we also have a separate route to handle loading by id.